clients/src/main/java/org/apache/kafka/common/record/AbstractRecords.java

```diff
@@ -126,11 +172,43 @@ protected LogRecord makeNext() {
         };
     }
 
-    public static int estimatedSizeRecords(CompressionType compressionType, Iterable<LogRecord> records) {
+    public static int sizeEstimateInBytes(byte magic,
+                                          long baseOffset,
+                                          CompressionType compressionType,
+                                          Iterable<LogRecord> records) {
         int size = 0;
-        for (LogRecord record : records)
-            size += record.sizeInBytes();
-        // TODO: Account for header overhead
+        if (magic <= LogEntry.MAGIC_VALUE_V1) {
+            for (LogRecord record : records)
+                size += record.sizeInBytes();
```

Comment:
Thanks for the explanation. That makes sense. Your call on which option to use then since you are most familiar with the hierarchies here.